tests/unit/test_acoustic_service.py

```diff
@@ -74,33 +74,55 @@ def test_ctms_to_acoustic_no_product(
 
 
 def test_ctms_to_acoustic_newsletters(
+    dbsession,
     base_ctms_acoustic_service,
-    minimal_contact,
+    email_factory,
     main_acoustic_fields,
-    acoustic_newsletters_mapping,
 ):
-    (main, newsletters, _) = base_ctms_acoustic_service.convert_ctms_to_acoustic(
-        minimal_contact, main_acoustic_fields, acoustic_newsletters_mapping
+    email = email_factory(newsletters=5)
+    email.newsletters[2].subscribed = False
+    email.newsletters[2].unsub_reason = "not interested"
+    dbsession.commit()
+
+    contact = ContactSchema.from_email(email)
+
+    acoustic_newsletters_mapping = {
+        contact.newsletters[0].name: "sub_column_1",
+        contact.newsletters[1].name: "sub_column_2",
+        contact.newsletters[2].name: "sub_column_3",
+    }
+
+    (main, newsletter_rows, _) = base_ctms_acoustic_service.convert_ctms_to_acoustic(
+        contact, main_acoustic_fields, acoustic_newsletters_mapping
     )
 
-    assert len(minimal_contact.newsletters) == 4
-    # Newsletters records
-    assert len(newsletters) == 2
-    app_dev, mofo = newsletters
-    assert (
-        app_dev["email_id"] == mofo["email_id"] == str(minimal_contact.email.email_id)
+    # As many records in the newsletters relation table as in contact
+    assert len(newsletter_rows) == len(contact.newsletters)
+    # `newsletter_name` for all names
+    assert sorted(nl["newsletter_name"] for nl in newsletter_rows) == sorted(
+        nl.name for nl in contact.newsletters
+    )
```

Comment:
I wish pytest had [`assertCountEqual`](https://docs.python.org/3.11/library/unittest.html#unittest.TestCase.assertCountEqual) for assertions like these.